### PR TITLE
[4.0] RavenDB-11446 Can't save index with whitespace

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Client/Documents/Indexes/PutIndexResult.cs [hunk 1]
```diff
@@ -4,13 +4,40 @@
...
+            return new DynamicJsonValue
+            {
+                [nameof(Index)] = Index
+            };
+        }
+    }
+    
+    public class PutIndexesResult : IDynamicJson
+    {
+        public List<PutIndexResult> indexes { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Indexes/PutIndexResult.cs]: Should be `Indexes`

File: src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs [hunk 2]
```diff
@@ -48,24 +48,25 @@ public class AdminIndexHandler : DatabaseRequestHandler
...
                 HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
 
+                var indexesResult = new PutIndexesResult
+                {
+                    indexes = createdIndexes
+                };
+                
                 using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
                 {
-                    writer.WriteStartObject();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs]: This is a breaking change. Old clients will not work properly. You need to stay with the JSON that we had.

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs [hunk 3]
```diff
@@ -21,14 +20,15 @@ public class AdminIndexHandler : DatabaseRequestHandler
...
-                var createdIndexes = new List<string>();
+                var createdIndexes = new List<PutIndexResult>();
                 var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Indexes");
                 if (input.TryGet("Indexes", out BlittableJsonReaderArray indexes) == false)
                     ThrowRequiredPropertyNameInRequest("Indexes");
 
                 foreach (var indexToAdd in indexes)
                 {
                     var indexDefinition = JsonDeserializationServer.IndexDefinition((BlittableJsonReaderObject)indexToAdd);
+                    indexDefinition.Name = indexDefinition.Name.Trim();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs]: This can throw NRE and name validation is done later in code.

Comment (ppekrol) [reply src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs]: You can change that to: `indexDefinition.Name?.Trim();`

Comment (Danielle9897) [reply src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs]: ok

File: src/Raven.Client/Documents/Indexes/PutIndexResult.cs (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Client/Documents/Indexes/PutIndexResult.cs]: class removed

File: src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs]: used ``` context.Write(writer, new DynamicJsonArray(createdIndexes.Select(x => x.ToJson())));``` instead

**Pull request merged**